packages/wrangler/src/pages/deploy.tsx

```diff
@@ -73,38 +75,65 @@ export function Options(yargs: CommonYargsArgv) {
 		});
 }
 
-export const Handler = async ({
-	_,
-	directory,
-	projectName,
-	branch,
-	commitHash,
-	commitMessage,
-	commitDirty,
-	skipCaching,
-	bundle,
-	noBundle,
-	config: wranglerConfig,
-}: PublishArgs) => {
+export const Handler = async (args: PagesDeployArgs) => {
+	let { branch, commitHash, commitMessage, commitDirty } = args;
+
 	// Check for deprecated `wrangler pages publish` command
-	if (_[1] === "publish") {
+	if (args._[1] === "publish") {
 		logger.warn(
 			"`wrangler pages publish` is deprecated and will be removed in the next major version.\nPlease use `wrangler pages deploy` instead, which accepts exactly the same arguments."
 		);
 	}
 
-	if (wranglerConfig) {
-		throw new FatalError("Pages does not support wrangler.toml", 1);
+	if (args.config) {
+		throw new FatalError(
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
+	}
+
+	/*
+	 * Check whether project contains a valid Pages `wrangler.toml` file. If it
+	 * does, we want to pick up the configuration specified in it. If it doesn't
+	 * we'll ignore the file, but inform users that we did find one, just not
+	 * valid for Pages.
+	 */
+	let config: Config | undefined;
+	const configPath = findWranglerToml(process.cwd(), false);
+	if (configPath) {
+		// this reads the config file with `env` set to `undefined`, which will
+		// return the top-level config. This contains all the information we
+		// need from now. We will perform a second config file read later
+		// in `/api/pages/deploy`, that will get the environment specififc config
+		config = readConfig(configPath, { ...args, env: undefined });
+		if (config.pages_build_output_dir === undefined) {
+			logger.warn(
```

Comment:
@dario-piotrowicz we kept this as a warning, but I changed the message based on our conversation. Let me know what you think of it <3